content/blog/esp32-memory-map-101/index.md

```diff
@@ -0,0 +1,351 @@
+
+This sparked the motivation to compile a visually rich and informative depiction of crucial information from the current versions of a TRM. This is intended to be used together with the SoC’s datasheets and TRM to help with a better understanding of the ESP32's family SoCs internals, which is crucial during the system and driver development - for example when porting to other platforms such as Zephyr RTOS.
+
+## ESP32
+
+The ESP32 is the Xtensa LX6 dual-core microcontroller, and it is the oldest SoC from the ESP32’s family of MCUs.
+
+The SoC has access to external Flash memory using the SPI interface. It can also access the external SPIRAM, but only on the data bus.
+
+Zephyr-RTOS support for the ESP32 devkit board can be found [here](https://docs.zephyrproject.org/latest/boards/espressif/esp32_devkitc_wrover/doc/index.html).
+
+
+**ROM**
+
+ROM memories hold the primary bootloader code and other library functions available to the end users. The content of the ROM memories is baked during the manufacturing process and cannot be changed. Please refer to the ECO versions of the ESP32 microcontroller for more detailed information. TLDR is to use the last ECOv3 and later ESP32 chip versions.
+
+**SRAM**
+
+The ESP32 has three internal SRAM modules. The SRAM0 is 192kB, and it can be configured to use the first 64kB as MMU cache for the external memory access. When not used as a cache, the first 64kB can be read and written by either CPU from the instruction bus.
+
+The capacity of Internal SRAM1 is 128 KB. Either CPU can read and write this memory using the data and instruction bus.
+
+The address ranges of the instruction and the data bus are in reverse order (word-wise).
+
+**RTC-SRAM**
+
+RTC-Fast Memory can be read and written by PRO_CPU only, at a given address range on the data bus or at the instruction bus. Unlike most other memory regions, RTC Fast memory cannot be accessed by the APP_CPU.
+
+RTC-Slow memory is r/w memory that can be accessed by either CPU via the instruction and data bus.
+
+The RTC memories can hold the data during the deep sleep period.
+
+**Instruction & data cache**
+
+The instruction and data cache address space is a virtual address space where the external memories such as Flash or SPIRAM modules can be accessed by the instruction or the data bus.
+
+
+### Internal memories and cache address space
+
+{{< figure
+    default=true
+    src="img/esp32-mmap.webp"
+    alt=""
+    caption="The ESP32 memory map."
+    >}}
+
+
+### Peripherals
+
+The ESP32 has 41 peripherals. Nearly all peripheral modules can be accessed by either CPU at the same address with a single exception of the PID controller which exists for each CPU and can be accessed only by its own CPU.
+
+{{< figure
```

Comment:
If I understand it correctly, the listed regions of memory contain registers for the respective peripherals.

Since you are deciphering TRM content here, and with the fact that registers are a big part of a TRM, suggest that you mention registers here.